Sources/SnapshottingTests/PreviewBaseTest.swift

```diff
@@ -70,12 +71,22 @@ open class PreviewBaseTest: XCTestCase {
       previews = []
       var i = 0
 
+      let currentDeviceName = ProcessInfo.processInfo.environment["SIMULATOR_DEVICE_NAME"] ?? ProcessInfo.processInfo.environment["SIMULATOR_MODEL_IDENTIFIER"]
+
       for discoveredPreview in discoveredPreviews {
         let typeName = discoveredPreview.typeName
         let displayName = discoveredPreview.displayName ?? typeName
         let count = discoveredPreview.numberOfPreviews
 
         for j in 0..<count {
+          // Filter out device specific previews whose device name doesn't match the currently selected one
+          if currentDeviceName != nil {
+            let specifiedPreviewDevice = discoveredPreview.devices[j]
+            guard specifiedPreviewDevice == "" || specifiedPreviewDevice == currentDeviceName else {
```

Comment:
It's a bit more Swifty to use the `isEmpty` var: https://developer.apple.com/documentation/swift/string/isempty